arch/sparc/core/switch.S

```diff
@@ -101,6 +105,15 @@ SECTION_FUNC(TEXT, z_sparc_context_switch)
 	nop
 	nop
 
+	/*
+	 * We have finished saving the "old" context and are also back in the
+	 * register window for which z_sparc_context_switch() was called.
+	 *
+	 * Now write the old thread into switch handle.
+	 * "old->switch_handle = old".
+	 */
+	st	%o1, [%o1 + ___thread_t_switch_handle_OFFSET]
```

Comment:
I would prefer to keep the update of the switch handle here so that the implementation matches the description in `kernel_arch_interface.h` as close as possible. The recent update to the `arch_switch()` description in that file was very helpful: Thanks!

Good point about the memory ordering. It is quite convenient on SPARC, because the specification defines a required "Total Store Ordering" memory model. One of the guarauntees is that the stores of all CPUs go to memory and are seen by other CPUs in the same order as the store instructions were issued.